apps/designer/app/designer/features/style-panel/sections/layout/layout.tsx

```diff
@@ -1,13 +1,241 @@
-import { Box, Flex, Grid } from "@webstudio-is/design-system";
+import { useState } from "react";
+import {
+  Box,
+  DeprecatedIconButton,
+  EnhancedTooltip,
+  Flex,
+  Grid,
+  Tooltip,
+} from "@webstudio-is/design-system";
+import type { StyleProperty, StyleValue } from "@webstudio-is/css-data";
 import { toValue } from "@webstudio-is/css-engine";
+import {
+  ColumnGapIcon,
+  RowGapIcon,
+  LinkedIcon,
+  UnlinkedIcon,
+} from "@webstudio-is/icons";
 import type { RenderCategoryProps } from "../../style-sections";
 import { FlexGrid } from "./shared/flex-grid";
-import { Lock } from "./shared/lock";
 import { renderProperty } from "../../style-sections";
-import { MenuControl, SelectControl, TextControl } from "../../controls";
+import { MenuControl, SelectControl } from "../../controls";
 import { PropertyName } from "../../shared/property-name";
-import { ColumnGapIcon, RowGapIcon } from "@webstudio-is/icons";
 import { styleConfigByName } from "../../shared/configs";
+import type { CreateBatchUpdate } from "../../shared/use-style-data";
+import { getStyleSource, type StyleInfo } from "../../shared/style-info";
+import {
+  type IntermediateStyleValue,
+  CssValueInput,
+} from "../../shared/css-value-input";
+
+const GapLinked = ({
+  isLinked,
+  onChange,
+}: {
+  isLinked: boolean;
+  onChange: (isLinked: boolean) => void;
+}) => {
+  return (
+    <Tooltip
+      content={isLinked ? "Unlink gap values" : "Link gap values"}
+      delayDuration={400}
+      disableHoverableContent={true}
+    >
+      <Flex
+        css={{
+          width: "100%",
+          justifyContent: "center",
+        }}
+      >
+        <DeprecatedIconButton onClick={() => onChange(!isLinked)}>
+          {isLinked ? <LinkedIcon /> : <UnlinkedIcon />}
+        </DeprecatedIconButton>
+      </Flex>
+    </Tooltip>
+  );
+};
+
+const GapInput = ({
+  icon,
+  style,
+  property,
+  intermediateValue,
+  setIntermediateValue,
+  onPreviewSet,
+  onPreviewDelete,
+  onSet,
```

Comment:
`setIntermediateValue` name is inconsistent with other, probably should be `onIntermediateValueChange`